src/lib.rs

```diff
@@ -431,7 +431,16 @@ macro_rules! redis_json_module_create {(
             export_shared_api(ctx);
             ctx.set_module_options(ModuleOptions::HANDLE_IO_ERRORS);
             ctx.log_notice("Enabled diskless replication");
-            $init_func(ctx, args)
+
+            let valid_version = unsafe{
+                rawmod::RedisModule_GetServerVersion.map_or(false, |get_version|0x060200 <= get_version())
```

Comment:
We don't allow redis 6.0 with ReJSON 2.x?

And what ever we decide here will affect all users of the macro `redis_json_module_create`
Including RedisJsonHdt
Can we make it to be another macro param (`min_version`)?

Reply:
RedisLabsModules/redismodule-rs#184 is the right, you're right